app/(aux)/settings.js

```diff
@@ -120,24 +230,227 @@ export default function Page() {
 		}
 	};
 
+	const handleChangeEmail = async () => {
+		if (newEmail !== confirmEmail) {
+			Alert.alert('Error', 'The new emails do not match.');
+			return;
+		}
+
+		let approvedEmailProviders = [
+			'gmail.com',
+			'yahoo.com',
+			'outlook.com',
+			'icloud.com',
+			'aol.com'
+		];
+
+		let confirmEmailCheck = confirmEmail.split('@');
+		if (confirmEmailCheck.length !== 2)
+			return alert('Invalid Email');
+		else if (!approvedEmailProviders.includes(confirmEmailCheck[1]))
+			return alert('Email provider not supported');
+
+		let me = await AsyncStorage.getItem('user');
+		writeData(`prayer_circle/users/${me}/private/email`, `${confirmEmail}`, true);
+
+		AsyncStorage.setItem('email', confirmEmail);
+
+		Alert.alert('Success', 'Email has been updated to: ' + confirmEmail);
+		bottomSheetModalRef.current?.dismiss();
+
+		auth.signOut();
+	};	
+
+	const handleEmptyCache = async () => {
+		AsyncStorage.clear();
+		Alert.alert('Success', 'Cache has been emptied.');
+		bottomSheetModalRef.current?.dismiss();
+
+		auth.signOut();
```

Comment:
I don't think this should sign the user out, or clear all LocalStorage. You could remove the bookmarked posts and timers but there is other data required for user data and such that we want to keep. This stuff should only be removed on sign out, not clearing the cache